Frontend: fix MetaMask false onboarding & add support for injected providers

GabrielBuragev opened the pull request:
Closes #1449 

Additionally adds support for any injected provider by a browser wallet extension.

Only tested with Frame Companion & MetaMask.

manuelwedler commented on frontend/src/composables/useWallet.ts:
```diff
@@ -28,6 +29,17 @@ export function useWallet(
     }
   }
 
+  async function connectInjected() {
```
A test for this would be nice too even if it is very basic behavior.

manuelwedler commented on frontend/tests/unit/services/web3-provider/injected-provider.spec.ts:
```diff
@@ -0,0 +1,103 @@
+  });
+
+  describe('InjectedProvider', () => {
+    describe('requestSigner()', () => {
```
Since the method moved to InjectedProvider, I think it is better to remove the respective tests from the MetaMask provider. Would mean we would have to maintain it in two places, if something changes.

GabrielBuragev replied:
Good point 👍

manuelwedler reviewed (changes requested):
Thanks! Looks mostly good. But if I didn't get something wrong the MetaMask onboarding is available at all anymore. Check the comment

GabrielBuragev commented on frontend/src/components/WalletMenu.vue:
```diff
@@ -101,19 +112,28 @@ const walletOptions = ref([
       walletOptions.value = walletOptions.value.filter((option) => option.hasMobileFlow);
     }
+  } else {
+    if (metaMaskAvailable) {
+      walletOptions.value = walletOptions.value.filter(
+        (option) => option.name !== 'Browser Wallet',
+      );
+    } else {
+      walletOptions.value = walletOptions.value.filter((option) => option.name !== 'MetaMask');
+    }
```
You are right. Nice catch!
I will just show MetaMask as an option when there is no injection detected since we only have onboarding support for MM currently.

GabrielBuragev commented on frontend/src/components/WalletMenu.vue:
```diff
@@ -73,10 +75,19 @@ const {
 
 const walletOptions = ref([
+  {
+    name: 'Browser Wallet',
```
Oh damn, somehow forgot to write tests. Will definitely do, thanks for the heads up!

GabrielBuragev commented on frontend/src/composables/useWallet.ts:
```diff
@@ -58,6 +70,9 @@ export function useWallet(
       case WalletType.Coinbase:
         return connectCoinbase();
+
+      case WalletType.Injected:
+        return connectInjected();
```
Indeed, this should be handled by our app. Will fix it.

GabrielBuragev commented:
On top of addressing your suggestions i used the opportunity to improve the tests for the `useWallet` composable.
Finished all the todos from there and removed some code duplication.

manuelwedler reviewed (approved):
Nice work. Thanks!
